src/activities/CreateMediaLayer.ts

```diff
@@ -40,13 +45,28 @@ export default class CreateMediaLayer implements IActivityHandler {
     execute(
         inputs: CreateMediaLayerInputs,
     ): CreateMediaLayerOutputs {
-        const { properties, source, title } = inputs;
+        let {source } = inputs;
+        const { properties, title } = inputs;
+
         if (!source) {
             throw new Error("source is required");
         }
 
+        if (!(source instanceof ImageElement ||
+            source instanceof LocalMediaElementSource ||
+            source instanceof VideoElement)) {
+
+            if ((source as __esri.ImageElementProperties).image) {
+                source = new ImageElement(source as __esri.ImageElementProperties);
+            } else if ((source as __esri.VideoElementProperties).video) {
+                source = new VideoElement(source as __esri.VideoElementProperties);
+            } else {
+                source = new LocalMediaElementSource(source as __esri.LocalMediaElementSourceProperties);
```

Comment:
Should we not also have an `if` condition for `source as __esri.LocalMediaElementSourceProperties` rather than assuming here?